types/src/vm/module_metadata.rs

```diff
@@ -211,13 +210,7 @@ pub fn get_metadata(md: &[Metadata]) -> Option<Arc<RuntimeModuleMetadataV1>> {
                 meta
             }
         })
-    } else {
-        get_metadata_v0(md)
-    }
-}
-
-pub fn get_metadata_v0(md: &[Metadata]) -> Option<Arc<RuntimeModuleMetadataV1>> {
-    if let Some(data) = md.iter().find(|md| md.key == APTOS_METADATA_KEY) {
+    } else if let Some(data) = md.iter().find(|md| md.key == APTOS_METADATA_KEY) {
```

Comment:
should be able to use find_metadata here and above?
I think we need better names for this method and ones below (see the other comment)